private/pkg/git/git.go

```diff
@@ -28,9 +28,30 @@ import (
 )
 
 const (
+	// DotGitDir is a relative path to the `.git` directory.
 	DotGitDir = ".git"
+
+	// ModeUnknown is a mode's zero value.
+	ModeUnknown FileMode = 0
+	// ModeFile is a blob that should be written as a plain file.
+	ModeFile FileMode = 010_0644
+	// ModeExec is a blob that should be written with the executable bit set.
+	ModeExe FileMode = 010_0755
+	// ModeDir is a tree to be unpacked as a subdirectory in the current
+	// directory.
+	ModeDir FileMode = 004_0000
+	// ModeSymlink is a blob with its content being the path linked to.
+	ModeSymlink FileMode = 012_0000
+	// ModeSubmodule is a commit that the submodule is checked out at.
+	ModeSubmodule FileMode = 016_0000
 )
 
+var ErrSubTreeNotFound = errors.New("subtree not found")
+
+// FileMode is how to interpret a tree node's object. See the Mode* constants
+// for how to interpret each mode value.
+type FileMode uint32
```

Comment:
```suggestion
type ObjectMode uint32
```

So when declaring `ModeFile FileMode = 010_0644` does not read as cryptic.